addons/survey/models/survey_survey.py

```diff
@@ -64,6 +64,7 @@ def default_get(self, fields_list):
         ('custom', 'Custom'),
     ],
         string='Survey Type', required=True, default='custom')
+    allowed_survey_types = fields.Json(string='Allowed survey types', compute="_compute_allowed_survey_types")
```

Comment:
Adding something... we could effectively add two computed fields to use in rules: one based on type, one based on restricted users. Would probably make rules easier to read (and then to update if necessary without having to do -u) and even join a lot of rules, as they would be computed in python. Just some wild proposal.
```
    is_survey_restriction_allowed = fields.Boolean()
    is_survey_type_allowed = fields.Boolean()

    def _compute_is_survey_restriction_allowed(self):
        # compute based on restricted users
        restricted = self.filtered('restrict_user_ids') if not user.has_group('manager') else self.browse()
        for survey in self:
            survey.is_survey_restriction_allowed = survey not in restricted or self.env.user in survey.restrict_user_ids

    def _compute_is_survey_type_allowed(self):
        # compute based on user groups
        restricted = self.filtered(lambda s: s.survey_type not in s.allowed_survey_types) if not user.has_group('manager') else self.browse()
        for survey in self:
            survey.is_survey_type_allowed = survey not in restricted
```

Reply:
Even if `('survey_type', 'in', allowed_survey_types)` technically works, it wouldn't be ideal because the right side of the domain will be cached when used in access rules. We found this with @Odoonan in eLearning not too long ago.

We can make however have something like `('can_access_survey_type', '=', True)`.

Then, it would somehow be simpler and we have used this pattern in other places, but are we sure we're OK with removing the users the ability to change these access rights from the UI?